apps/framework-cli/src/framework/schema.rs

```diff
@@ -378,10 +408,152 @@ pub fn ast_mapper(ast: SchemaAst) -> Result<FileObjects, ParsingError> {
     Ok(FileObjects::new(parsed_models, enums))
 }
 
+pub fn ts_ast_mapper(ast: Module) -> Result<FileObjects, ParsingError> {
+    let models = Vec::new();
+    let mut enums = Vec::new();
+
+    let mut ts_declarations = Vec::new();
+
+    // collect all inteface and enum declarations
+    ast.body.iter().for_each(|item| match item {
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsInterface(decl))) => {
+            ts_declarations.push(decl);
+        }
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsEnum(decl))) => {
+            enums.push(ts_enum_to_data_enum(decl));
+        }
+        _ => {}
+    });
+
+    println!(
+        "the interfaces {:#?}",
+        ts_interface_to_model(ts_declarations[0], &enums)
+    );
+
+    Ok(FileObjects::new(models, enums))
+}
+
+fn ts_enum_to_data_enum(enum_decl: &TsEnumDecl) -> DataEnum {
+    let name = enum_decl.id.sym.to_string();
+    let values = enum_decl
+        .members
+        .iter()
+        .map(|member| match member {
+            TsEnumMember {
+                id: TsEnumMemberId::Ident(ident),
+                ..
+            } => ident.sym.to_string(),
+            _ => "unsupported".to_string(),
+        })
+        .collect();
+    DataEnum { name, values }
+}
+
+fn ts_interface_to_model(
+    interface: &TsInterfaceDecl,
+    enums: &[DataEnum],
+) -> Result<DataModel, ParsingError> {
+    let schema_name = interface.id.sym.to_string();
+
+    let columns: Result<Vec<Column>, ParsingError> = interface
+        .body
+        .body
+        .iter()
+        .filter_map(|field| match field {
+            swc_ecma_ast::TsTypeElement::TsPropertySignature(prop) => Some({
+                // match the key's sym if it's an ident
+                let name = match *prop.key.clone() {
+                    Expr::Ident(ident) => ident.sym.to_string(),
+                    _ => return None,
+                };
+                // match the type of the value and return the right column type
+                let TsTypeAnn { type_ann, .. } =
+                    *prop.type_ann.clone().expect("no type for property");
```

Comment:
This is going to panic if the user fails to put a type. They will do that eventually, maybe we solidify this in a follow up PR with error handling.